### ISSUE TITLE
Can't get past "LayerNormKernelImpl" not implemented for 'Half'

### ISSUE THREAD

**Issue by offmybach** [comfyanonymous/ComfyUI#2938]
I've been stuck for a day not not being able to run a wf because of these 'half' issues. I'm using Stability Matrix and errors out every time, I have force fp 32 argument and box check for force fp32. I've also gotten mat1 and mat 2 must match errors ,and  other half errors besides the layernormkernal..Error occurred when executing CLIPTextEncode:

"LayerNormKernelImpl" not implemented for 'Half'

  File "D:\Data\Packages\ComfyUI\execution.py", line 152, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "D:\Data\Packages\ComfyUI\execution.py", line 82, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "D:\Data\Packages\ComfyUI\execution.py", line 75, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "D:\Data\Packages\ComfyUI\nodes.py", line 56, in encode
    cond, pooled = clip.encode_from_tokens(tokens, return_pooled=True)
  File "D:\Data\Packages\ComfyUI\comfy\sd.py", line 135, in encode_from_tokens
    cond, pooled = self.cond_stage_model.encode_token_weights(tokens)
  File "D:\Data\Packages\ComfyUI\comfy\sd1_clip.py", line 518, in encode_token_weights
    out, pooled = getattr(self, self.clip).encode_token_weights(token_weight_pairs)
  File "D:\Data\Packages\ComfyUI\comfy\sd1_clip.py", line 39, in encode_token_weights
    out, pooled = self.encode(to_encode)
  File "D:\Data\Packages\ComfyUI\comfy\sd1_clip.py", line 195, in encode
    return self(tokens)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\comfy\sd1_clip.py", line 177, in forward
    outputs = self.transformer(tokens, attention_mask, intermediate_output=self.layer_idx, final_layer_norm_intermediate=self.layer_norm_hidden_state)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\comfy\clip_model.py", line 134, in forward
    x = self.text_model(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\comfy\clip_model.py", line 109, in forward
    x, i = self.encoder(x, mask=mask, intermediate_output=intermediate_output)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\comfy\clip_model.py", line 68, in forward
    x = l(x, mask, optimized_attention)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\comfy\clip_model.py", line 49, in forward
    x += self.self_attn(self.layer_norm1(x), mask, optimized_attention)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\comfy\ops.py", line 108, in forward
    return self.forward_comfy_cast_weights(*args, **kwargs)
  File "D:\Data\Packages\ComfyUI\comfy\ops.py", line 104, in forward_comfy_cast_weights
    return torch.nn.functional.layer_norm(input, self.normalized_shape, weight, bias, self.eps)
  File "D:\Data\Packages\ComfyUI\venv\lib\site-packages\torch\nn\functional.py", line 2543, in layer_norm
    return torch.layer_norm(input, normalized_shape, weight, bias, eps, torch.backends.cudnn.enabled)

**Comment by wocall** [comment]
How was it ultimately resolved?
